### PR TITLE
[WIP] networking: Calico networking support

### PR DESCRIPTION
**Pull request by jellonek**
Calico networking implementation for #33 
- [x] add calico daemons into package
- [x] tap device creation (save info about it somewhere in sandbox)
- [x] inform calico about new workload/endpoint
- [x] pass tap device to libvirt as network device
- [x] pass addressing data from calico in PodSandboxStatus

---

This change is [Reviewable](https://reviewable.io/reviews/mirantis/virtlet/46)

### CODE REVIEW

**Comment by jellonek** [comment]
Instead providing own installation of calico - operator is asked in `README.me` to prepare it on each node using instructions from https://github.com/projectcalico/calico-containers/blob/master/docs/cni/kubernetes/KubernetesIntegration.md

File: pkg/utils/tuntap.go [hunk 1]
```diff
@@ -0,0 +1,98 @@
...
+	iface, err := Open(name, kind)
+	if err != nil {
+		return "", err
+	}
+	defer iface.Close()
+
+	_, _, errno := syscall.Syscall(syscall.SYS_IOCTL, iface.file.Fd(), uintptr(syscall.TUNSETPERSIST), persistency)
+
+	if errno != 0 {
+		return "", fmt.Errorf("ioctl failed (TUNSETPERSIST): %v", errno)
```
Comment (nhlfr) [inline pkg/utils/tuntap.go]: Please convert errno to the Go error by using `syscall.Errno`, like here:
https://github.com/Mirantis/virtlet/blob/master/pkg/libvirttools/image.go#L147

Comment (jellonek) [inline pkg/utils/tuntap.go]: This errno is already of type `syscall.Errno` - ptal https://golang.org/pkg/syscall/#Syscall

Comment (nhlfr) [inline pkg/utils/tuntap.go]: ah, ok

**Comment by jellonek** [comment]
Replaced by #126 and #130

**Pull request closed without merging**